### PR TITLE
feat: Support for CXML self service configuration

### PR DESCRIPTION
**Pull request by suschneider**
## PR Type

[x] Feature

## What Is the Current Behavior?

It is not possible to configure specific configurations for CXML Punchout users in the PWA.

## What Is the New Behavior?

User specific configurations for CXML Punchout are possible in the MyAccount area of the PWA.

## Does this PR Introduce a Breaking Change?

[x] No

## Other Information

new punchout REST API available with ICM12.2

IAD: https://www.figma.com/design/DnogQvuEuB0fbqs7uG264R/CXML-self-service-configuration-in-MyAccount?node-id=0-1&t=yaKr7p89rS9FLcBm-0

[AB#95963](https://dev.azure.com/intershop-com/Products/_workitems/edit/95963)

[AB#97411](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/97411)

### CODE REVIEW

File: src/assets/i18n/de_DE.json [hunk 1]
```diff
@@ -360,9 +360,15 @@
   "account.punchout.configuration.link": "Konfiguration",
   "account.punchout.configuration.option.none.label": "Keine",
   "account.punchout.configuration.save_success.message": "Die OCI-Punchout-Konfiguration wurde gespeichert.",
+  "account.punchout.configure.link": "Konfigurieren {{0}}",
```
Comment (mglatter) [inline src/assets/i18n/de_DE.json:363]:
```suggestion
  "account.punchout.configure.link": "{{0}} konfigurieren",
```

File: src/assets/i18n/de_DE.json [hunk 2]
```diff
@@ -360,9 +360,15 @@
...
   "account.punchout.create.description": "Bitte geben Sie die Anmeldedaten des {{0, translate, account.punchout.type.text}}-Punchout-Benutzers an.",
   "account.punchout.create.heading": "Einen neuen {{0, translate, account.punchout.type.text}}-Punchout-Benutzer anlegen",
   "account.punchout.create.link": "Einen neuen Punchout-Benutzer anlegen",
+  "account.punchout.cxml.configuration.default.description": "Standardwert: {{defaultValue}}",
+  "account.punchout.cxml.configuration.helptext": "Verwenden Sie die folgenden Einstellungen, um den Inhalt des cXML-Punchout-Dokuments festzulegen. Eingestellte Werte können zurückgesetzt werden, um auf den angegebenen Standardwert zurückzukehren.",
```
Comment (mglatter) [inline src/assets/i18n/de_DE.json:368]:
```suggestion
  "account.punchout.cxml.configuration.helptext": "Verwenden Sie die folgenden Einstellungen, um den Inhalt des cXML-Punchout-Dokuments festzulegen. Eingestellte Werte können auf den angegebenen Standardwert zurückgesetzt werden.",
```

**Review by marschmidt89 — APPROVED** [review]
feat: added suggestions

File: src/assets/i18n/en_US.json [hunk 3]
```diff
@@ -360,9 +360,15 @@
...
-  "account.punchout.create.heading": "Create new {{0, translate, account.punchout.type.text}} punchout user",
-  "account.punchout.create.link": "Create new punchout user",
+  "account.punchout.create.heading": "Create a new {{0, translate, account.punchout.type.text}} punchout user",
+  "account.punchout.create.link": "Create a new punchout user",
+  "account.punchout.cxml.configuration.default.description": "Default value: {{defaultValue}}",
+  "account.punchout.cxml.configuration.helptext": "Use the following preferences to determine the content of the cXML Punchout document. Set values can be reset to return to the default value as specified.",
+  "account.punchout.cxml.configuration.link": "Configuration",
+  "account.punchout.cxml.configuration.no_configuration": "There is no configuration available at this time.",
+  "account.punchout.cxml.configuration.save_success.message": "The cXML Punchout configuration has been saved.",
   "account.punchout.cxml.info.url.helptext": "Please use the following URL for the punchout configuration in your cXML procurement system. For the sender credentials use the username as identity and the password as SharedSecret.",
```
Comment (mglatter) [inline src/assets/i18n/en_US.json:372]: Identity and SharedSecret are in quotation marks in the other languages. Not needed in EN?

**Review by SGrueber — CHANGES_REQUESTED** [review]

File: src/app/extensions/punchout/services/punchout/punchout.service.ts [hunk 4]
```diff
@@ -299,6 +302,62 @@ export class PunchoutService {
...
+   * @returns       An array of punchout cXML configurations.
+   */
+  getCxmlConfiguration(user: string): Observable<CxmlConfiguration[]> {
+    return this.currentCustomer$.pipe(
+      switchMap(customer =>
+        this.apiService
+          .get<CxmlConfigurationData>(
+            `customers/${encodeResourceID(customer.customerNo)}/punchouts/${encodeResourceID(
+              this.getResourceType('cxml')
+            )}/users/${user}/configurations`,
```
Comment (SGrueber) [inline src/app/extensions/punchout/services/punchout/punchout.service.ts:318]: add encodeResourceId for the userId

File: src/app/extensions/punchout/services/punchout/punchout.service.ts [hunk 5]
```diff
@@ -299,6 +302,62 @@ export class PunchoutService {
...
+      return throwError(() => new Error('updateCxmlConfiguration() called without required CxmlConfiguration'));
+    }
+
+    return this.currentCustomer$.pipe(
+      switchMap(customer =>
+        this.apiService
+          .patch<CxmlConfigurationData>(
+            `customers/${encodeResourceID(customer.customerNo)}/punchouts/${encodeResourceID(
+              this.getResourceType('cxml')
+            )}/users/${user}/configurations`,
```
Comment (SGrueber) [inline src/app/extensions/punchout/services/punchout/punchout.service.ts:348]: add encodeResourceId for the userId

File: src/app/extensions/punchout/services/punchout/punchout.service.ts [hunk 6]
```diff
@@ -299,6 +302,62 @@ export class PunchoutService {
...
+  updateCxmlConfiguration(cxmlConfiguration: CxmlConfiguration[], user: string): Observable<CxmlConfiguration[]> {
+    if (!cxmlConfiguration) {
+      return throwError(() => new Error('updateCxmlConfiguration() called without required CxmlConfiguration'));
+    }
+
+    return this.currentCustomer$.pipe(
+      switchMap(customer =>
+        this.apiService
+          .patch<CxmlConfigurationData>(
+            `customers/${encodeResourceID(customer.customerNo)}/punchouts/${encodeResourceID(
```
Comment (SGrueber) [inline src/app/extensions/punchout/services/punchout/punchout.service.ts:346]: the encoding of the resourceType is not necessary even this function is used elsewhere

**Review by SGrueber — CHANGES_REQUESTED** [review]

**Review by mglatter — APPROVED** [review]

**Pull request merged**